### PR TITLE
Static release

### PR DESCRIPTION
**Pull request by DolceTriade**
Allows for easier Linux static releases.

### CODE REVIEW

File: daemon/external_deps/build.sh [hunk 1]
```diff
@@ -211,8 +213,11 @@ build_sdl2() {
 		cp -R "SDL2.framework" "${PREFIX}"
 		;;
 	*)
-		echo "Unsupported platform for SDL2"
-		exit 1
+		download "SDL2-${SDL2_VERSION}.tar.gz" "https://www.libsdl.org/release/SDL2-${SDL2_VERSION}.tar.gz" sdl2
```
Comment (mbasaglia) [inline daemon/external_deps/build.sh]: Wouldn't it better to use an explicit platform string instead of *?

**Comment by Kangz** [comment]
LGTM
Would be nice to have only explicit platforms yeah.

File: daemon/external_deps/build.sh (hunk 1 above)
Comment (DolceTriade) [inline daemon/external_deps/build.sh]: Done.

**Comment by DolceTriade** [comment]
PTAL

File: daemon/external_deps/build.sh [hunk 2]
```diff
@@ -114,8 +115,9 @@ build_zlib() {
 		make -f win32/Makefile.gcc install BINARY_PATH="${PREFIX}/bin" LIBRARY_PATH="${PREFIX}/lib" INCLUDE_PATH="${PREFIX}/include" SHARED_MODE=1
 		;;
 	*)
-		echo "Unsupported platform for zlib"
-		exit 1
+		./configure --prefix="${PREFIX}" --static --const
+		make
+		make install
```
Comment (mbasaglia) [inline daemon/external_deps/build.sh]: Here as well? :stuck_out_tongue_winking_eye:

File: daemon/external_deps/build.sh [hunk 3]
```diff
@@ -232,8 +237,8 @@ build_glew() {
 		install_name_tool -id "@rpath/libGLEW.${GLEW_VERSION}.dylib" "${PREFIX}/lib/libGLEW.${GLEW_VERSION}.dylib"
 		;;
 	*)
-		echo "Unsupported platform for GLEW"
-		exit 1
+		make GLEW_DEST="${PREFIX}"
+		make install GLEW_DEST="${PREFIX}"
```
Comment (mbasaglia) [inline daemon/external_deps/build.sh]: And here :open_mouth:

File: daemon/external_deps/build.sh [hunk 4]
```diff
@@ -303,16 +308,21 @@ build_openal() {
...
 	*)
-		echo "Unsupported platform for OpenAL"
-		exit 1
+		download "openal-soft-${OPENAL_VERSION}.tar.bz2" "http://kcat.strangesoft.net/openal-releases/openal-soft-${OPENAL_VERSION}.tar.bz2" openal
+		cd "openal-soft-${OPENAL_VERSION}"
+		cmake -DCMAKE_INSTALL_PREFIX="${PREFIX}" -DALSOFT_EXAMPLES=OFF -DLIBTYPE=STATIC .
+		make
+		make install
+		echo -ne "create libopenal-combined.a\naddlib libopenal.a\naddlib libcommon.a\nsave\nend\n" | ar -M
+		cp "libopenal-combined.a" "${PREFIX}/lib/libopenal.a"
```
Comment (mbasaglia) [inline daemon/external_deps/build.sh]: :arrow_up:

File: daemon/external_deps/build.sh (hunk 2 above)
Comment (DolceTriade) [inline daemon/external_deps/build.sh]: Done

File: daemon/external_deps/build.sh (hunk 3 above)
Comment (DolceTriade) [inline daemon/external_deps/build.sh]: Done

File: daemon/external_deps/build.sh (hunk 4 above)
Comment (DolceTriade) [inline daemon/external_deps/build.sh]: Done

File: daemon/external_deps/build.sh [hunk 5]
```diff
@@ -113,8 +114,13 @@ build_zlib() {
...
 		make -f win32/Makefile.gcc install BINARY_PATH="${PREFIX}/bin" LIBRARY_PATH="${PREFIX}/lib" INCLUDE_PATH="${PREFIX}/include" SHARED_MODE=1
 		;;
+	linux*)
+		./configure --prefix="${PREFIX}" --static --const
+		make
+		make install
+		;;
 	*)
-		echo "Unsupported platform for zlib"
+		echo "Unsupportd platform for zlib"
```
Comment (mbasaglia) [inline daemon/external_deps/build.sh]: Typo

Comment (DolceTriade) [inline daemon/external_deps/build.sh]: done.

**Comment by mbasaglia** [comment]
Looks good to me

**Pull request merged**